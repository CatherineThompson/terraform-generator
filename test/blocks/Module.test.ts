import { Module } from '../..';
import { tfGenerator11, tfGenerator12, arg4 } from '.';

test('Module 0.11', () => {
  const module = new Module(tfGenerator11, 'name', arg4(tfGenerator11));
  expect(module.toTerraform()).toMatchSnapshot();
  expect(module.asArgument().toTerraform()).toMatchSnapshot();
  expect(module.getAttribute('attr').toTerraform()).toMatchSnapshot();
});

test('Module 0.12', () => {
  const module = new Module(tfGenerator12, 'name', arg4(tfGenerator12));
  expect(module.toTerraform()).toMatchSnapshot();
  expect(module.asArgument().toTerraform()).toMatchSnapshot();
  expect(module.getAttribute('attr').toTerraform()).toMatchSnapshot();
});
