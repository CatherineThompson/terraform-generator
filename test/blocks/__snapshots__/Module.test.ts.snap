// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module 0.11 1`] = `
"module \\"name\\"{
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
}
arg12 = {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
}
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
}
}
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
}
}
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
}
}
}
}

"
`;

exports[`Module 0.11 2`] = `"module.name"`;

exports[`Module 0.11 3`] = `"module.name.attr"`;

exports[`Module 0.12 1`] = `
"module \\"name\\"{
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
}
arg12 = {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
}
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
}
}
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
}
}
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
}
}
}
}

"
`;

exports[`Module 0.12 2`] = `"module.name"`;

exports[`Module 0.12 3`] = `"module.name.attr"`;
